app/javascript/packages/document-capture/components/form-steps.jsx

```diff
@@ -107,6 +118,19 @@ function FormSteps({ steps = [], onComplete = () => {}, initialValues = {}, init
   const headingRef = useRef(/** @type {?HTMLHeadingElement} */ (null));
   const [stepName, setStepName] = useHistoryParam('step', initialStep);
   const { t } = useI18n();
+  const fields = useRef(/** @type {Record<string,HTMLElement?>} */ ({}));
+  useMemo(() => {
+    fields.current = {};
+  }, [stepName]);
+  const didSubmitWithErrors = useRef(false);
+  useEffect(() => {
+    if (activeErrors && didSubmitWithErrors.current) {
+      const firstActiveError = Object.keys(activeErrors)[0];
```

Comment:
is `activeErrors` in a useful order?

Reply:
> is `activeErrors` in a useful order?

Yes and no.

No in that relying on object keys [may or may not be reliable](https://stackoverflow.com/a/30919039).

Yes in that, in real-world JavaScript runtimes, they follow the order they're assigned, and we (currently) assign them in the order we expect:

https://github.com/18F/identity-idp/blob/246a019f68c20eaaa69cf55d2d26c58e2051cf7a/app/javascript/packages/document-capture/components/documents-step.jsx#L76-L82

To have stronger guarantees about order, we could refactor this to return an array of `{ fieldName, error }` from `validate`, similar to what we're considering for the REST API response.

Reply:
I restructured the error to an array of `{ fieldName, error }` in 8f021a3, which should also benefit us for work-in-progress work to structure server-received errors the same way.